Update release notes for v6 24 00 (#7866)
root-project/root: README/ReleaseNotes/v624/index.md

- Due to some planned major improvements to `RVec`, the layout of `RVec` objects will change in a backward-incompatible way between v6.24 and v6.26.
  Because of this, we now print a warning if an application is reading or writing a `ROOT::RVec` object from/to a ROOT file. We assume this is an
  exceedingly rare case, as the ROOT interface typically used to manipulate `RVec`s is `RDataFrame`, and `RDataFrame` performs an on-the-fly

>   `RVec <-> std::vector` conversion rather than writing `RVec`s to disk. Note that, currently, `RVecs` written e. Enable autig. in a `TTree` cannot be read back

Reviewer: This seems broken?
```suggestion
  `RVec <-> std::vector` conversion rather than writing `RVec`s to disk. Note that, currently, `RVecs` written e.g. in a `TTree` cannot be read back
```

  using certain ROOT interfaces (e.g. `TTreeReaderArray`, `RDataFrame` and the experimental `RNTuple`). All these limitations will be lifted in v6.26.
- Portable implementation of the RANLUX++ generator, see [RanluxppEngine](https://root.cern/doc/master/classROOT_1_1Math_1_1RanluxppEngine.html) and [our blog post](https://root.cern/blog/ranluxpp/).
- Change `TRandom3::GetSeed` to return the current state element in the contained seed vector of TRandom3. The return value will now change after every call of `TRandom3::Rndm` (when generating a random number). Before the function was returning the first element of the state, which was changing only after 624 calls to `Rndm()`.
- Fix a bug in `ROOT::Fit::BinData` copy constructor
- Fix a bug in applying a correction factor used for the computation of the fit confidence level in `ROOT::Fit::FitResult`.
- TMatrix: optimize implementation of `TPrincipal::AddRow` that is heavily used by CMS.

> ###Minuit2

Reviewer: ```suggestion
### Minuit2
```

- Add a new improved message logging system. Debug message now can be enabled in Minuit2 when using maximum print level.
- When using external provided gradient, compute in MnSeed still numerical gradients to obtain correct step sizes and  initial estimate of covariance matrix. This allows to start with a good first
	state estimation, reducing significantly the number of  subsequent iterations.

## TMVA

> - Introducing TMVA PyTorch Interface, a method to use PyTorch internally with TMVA for deep learning. This can be used as an alternative to PyKeras Interface for complex models providing more
> flexibility and power.

Reviewer: ```suggestion
- Introducing TMVA PyTorch Interface, a method to use PyTorch internally with TMVA for deep learning. This can be used as an alternative to PyKeras Interface for complex models providing more flexibility and power.
```

  These points will be skipped, and HypoTestInverter can continue.
- Tweak pull / residual plots. ROOT automatically zoomed out a bit when a pull / residual plot is created. Now, the
  axis range of the original plot is transferred to the residual plot, so the pulls can be drawn below the main plot.
- Improve plotting of `RooBinSamplingPdf`
- Print a Warning message when the `RooAddPdf` is evaluated without passing a normalization set and the class has not a normalization set defined.

> Without a normalization set the `RooAddPdf` is not properly defined and its shape will be different depending on which normalization range is used.

Reviewer: ```suggestion
 Without a normalization set the `RooAddPdf` is not properly defined and its shape will be different depending on which normalization range is used.
```

- Add the method `AddPoint`to `TGraph(x,y)` and `TGraph2D(x,y,z)`, equivalent to `SetPoint(g->GetN(),x,y)`and `SetPoint(g->GetN(),x,y,z)`
- Option `E0` draws error bars and markers are drawn for bins with 0 contents. Now, combined

> with options E1 and E2, it avoids error bars clipping.

Reviewer: ```suggestion
  with options E1 and E2, it avoids error bars clipping.
```
Else this ends the enumeration
